lib/pacemaker/pcmk_fence.c

```diff
@@ -76,23 +76,23 @@ handle_level(stonith_t *st, char *target, int fence_level,
 static void
 notify_callback(stonith_t * st, stonith_event_t * e)
 {
-    if (e->result != pcmk_ok) {
-        return;
-    }
+    if (pcmk__str_eq(async_fence_data.target, e->target, pcmk__str_casei)
+        && pcmk__str_eq(async_fence_data.action, e->action, pcmk__str_casei)) {
```

Comment:
for action we could go for case-sensitive comparison as we are touching it.

Reply:
good catch, thanks

Reply:
It might make sense to have some further wrappers for this stuff (they could just be macros) like pcmk__action_eq, pcmk__node_eq, etc. that encode the case sensitive vs. case insensitive knowledge somewhere.

Reply:
I was thinking about that too. However they may proliferate beyond our ability to remember they exist -- especially considering we could have e.g. node_eq wrappers for const char*, pe_node_t*, and crm_node_t*

Reply:
Oh yeah, plus we would never finish updating code to use them.  I guess they could all be defined in a single place to keep it easier to keep track of.  I was just thinking how I can't possibly remember what should be compared taking case into account.